src/service/account/index.ts

```diff
@@ -1,34 +1,35 @@
+import { AuthenticationError } from "../../common/errors";
+import { Connection } from "../../connection";
 import { Context } from "../../types";
-import { ACCOUNT, ACCOUNT_BY_NAME } from "../../common/api";
 
 export class AccountService {
   private context: Context;
 
   id!: string;
+  name!: string;
+  connection!: Connection;
 
   constructor(context: Context) {
     this.context = context;
   }
 
-  async resolveAccountId(account?: string) {
-    const { httpClient, apiEndpoint } = this.context;
-    if (account) {
-      const queryParams = new URLSearchParams({ account_name: account });
-      const url = `${apiEndpoint}/${ACCOUNT_BY_NAME}?${queryParams}`;
-      const { account_id } = await httpClient
-        .request<{ account_id: string }>("GET", url)
-        .ready();
-      this.id = account_id;
-      return account_id;
-    } else {
-      const url = `${apiEndpoint}/${ACCOUNT}`;
-      const {
-        account: { id }
-      } = await httpClient
-        .request<{ account: { id: string } }>("GET", url)
-        .ready();
-      this.id = id;
-      return id;
+  // TODO: find a unified place for this
+  private throwErrorIfNoConnection() {
```

Comment:
can we make `connection` required parameter for `resourceManager`?
We expect to use only engine connection in resource manager, and no more api calls.

Reply:
maybe this will help to remove all this `throwErrorIfNoConnection` calls?

Reply:
The idea was to keep disruption for the user's workflow to a minimum. In order to support standalone resource manager instantiation I had to make the connection optional, as RM has credentials to establish the connection only after the user calls `rm.authenticate()`. Though removing authenticate altogether and forcing users to provide the creds in constructor will make this much simpler...

Reply:
Discussed with the team, since we're releasing a major version might as well change the interface slightly, if this allows us to simplify the code.